tests/integration/ha_tests/application-charm/src/charm.py

```diff
@@ -72,12 +89,23 @@ def _on_start(self, _) -> None:
         """Only sets an Active status."""
         self.unit.status = ActiveStatus()
 
-    def _on_database_created(self, _) -> None:
-        """Event triggered when a database was created for this application."""
-        self._start_continuous_writes(1)
-
     def _on_endpoints_changed(self, _) -> None:
         """Event triggered when the read/write endpoints of the database change."""
+        if self._connection_string is None:
+            return
+
+        if not self.app_peer_data.get(PROC_PID_KEY):
+            return None
```

Comment:
Do you really need to return `None` at this point?

Reply:
If there's no `continuous_writes` running, because it was stopped or because it has crashed, we shouldn't be restarting it.

Reply:
feel free to ignore
```suggestion
            return
```